Pokemon Emerald: Link to Rom Changes in Game Page

nicholassaylor opened the pull request:
## What is this fixing or adding?
~Adds rematches being disabled to the list on the game page of game changes~
Links to the comprehensive list of rom changes in the game page so that full details can be found

## How was this tested?
Looking at renders

Resolves #3959

palex00 commented:
@Zunawe

Exempt-Medic commented on worlds/pokemon_emerald/docs/en_Pokemon Emerald.md:
```diff
@@ -30,6 +30,7 @@ randomizer. Here are some of the more important ones:
 - The Wally catching tutorial is skipped
 - All text is instant and, with an option, can be automatically progressed by holding A
 - When a Repel runs out, you will be prompted to use another
+- Trainers no longer ask to register you for rematches after a battle
```
I think the wording here is ambiguous and could be read as them _automatically_ registering instead

Exempt-Medic reviewed (commented):
I'm still not sure the wording is really "correct". As shown in the video in the bug report, the registration is still there, it just doesn't function properly

nicholassaylor commented:
> I'm still not sure the wording is really "correct". As shown in the video in the bug report, the registration is still there, it just doesn't function properly

For some reason, I can't view the videos. They are marked as corrupt by Github. 
![image](https://github.com/user-attachments/assets/b37496b5-6f17-4168-92d6-60594442bbd0)

I think I got closer to what we're looking for though. Wording might still be slightly off

Exempt-Medic commented on worlds/pokemon_emerald/docs/en_Pokemon Emerald.md:
```diff
@@ -30,6 +30,7 @@ randomizer. Here are some of the more important ones:
 - The Wally catching tutorial is skipped
 - All text is instant and, with an option, can be automatically progressed by holding A
 - When a Repel runs out, you will be prompted to use another
+- Rematches with trainers have been disabled, even if the trainer is registered in the PokeNav
```
Minor correction, but otherwise this LGTM
```suggestion
- Rematches with trainers have been disabled, even if the trainer is registered in the PokéNav
```

Exempt-Medic reviewed (approved):
One small comment, LGTM

Zunawe commented:
Didn't realize I missed that issue.

The thing is, this is labeled as a list of "quality of life improvements", and it ends with "Many more minor improvements". This piece of information feels out of place. It _should_ actually be in `rom changes.md` (did I really put spaces in these file names?), but it's missing from there. So if nothing else, this PR could add that line under Misc for that document.

If I could reasonably link to that document from the game info page, I would, but as far as I know there's not really a way to get it on the site, and there's no way to format a GitHub link that goes to "this file at the tag for the most recent release". I guess it's possible to put it in the guides, lol, but I don't think that's the right place for it.

nicholassaylor commented:
Currently, the change *is* in the `rom changes.md` https://github.com/ArchipelagoMW/Archipelago/blob/4dde3a219179d789182da1ca72c950ab9705f064/worlds/pokemon_emerald/docs/rom%20changes.md?plain=1#L16

The issue is that it is not evident that this file has information, so I am adding a link to the document which can be displayed on the webhost. (See DS3 for another example). In order to make it work, the file would need to be appended with "_en" for it to be renderable on the site, so I renamed the file and replaced the space with an underscore since you seemed to dislike it.

Zunawe commented:
Ah, you're right, I just searched pokegear and gave up.

I suppose I was under the impression that it needed to get a section in the setup page, but if it can be added "silently" and linked to, that sounds perfect.

Zunawe reviewed (approved):
Looks fine. Clicked the link, and it went to the page. It didn't have an unexpected entry on the setup guides page.

![Capture](https://github.com/user-attachments/assets/a17370cc-d2e9-4dd6-96cc-07f51754b0f4)
